pandas/core/series.py

```diff
@@ -3115,10 +3128,9 @@ def _sanitize_index(data, index, copy=False):
 
     if isinstance(data, ABCIndexClass) and not copy:
         pass
-    elif isinstance(data, PeriodIndex):
-        data = data.astype(object).values
-    elif isinstance(data, DatetimeIndex):
-        data = data._to_embed(keep_tz=True)
+    elif isinstance(data, (PeriodIndex, DatetimeIndex)):
```

Comment:
can you add a comment. This interface is now really smelly. I thought the purpose of ``._values`` was to do this on ALL index classes? to return the 'data' in a dtype appropraite manner? if that is not the case then this needs fixing.

Reply:
What's the issue here? That's exactly what `._values` did. I changed the two elifs into the one.

I can't remove any more since the `if` branch is dealing with copys and the `else` goes on to do more stuff in `_sanitize_arrary`.

Reply:
`_to_embed` previously took a copy, while now `data._values` does not? (but don't know if this is needed in this context here)

Reply:
Let's just follow `copy`.